src/tests/audio/tasks/forced_alignment_test.py

```diff
@@ -0,0 +1,215 @@
+
+
+@pytest.fixture
+def mono_audio_sample() -> Audio:
+    """Fixture for sample mono audio."""
+    return Audio.from_filepath(MONO_AUDIO_PATH)
+
+
+class SingleCharSegment(TypedDict):
+    """A single char of a speech."""
+
+    char: str
+    start: Optional[float]
+    end: Optional[float]
+    score: float
+
+
+@pytest.fixture
+def dummy_segment() -> SingleSegment:
+    """Fixture for a dummy segment."""
+    return SingleSegment(
+        start=0.0,
+        end=1.0,
+        text="test",
+        clean_char=["t", "e", "s", "t"],
+        clean_cdx=[0, 1, 2, 3],
+        clean_wdx=[0],
+        sentence_spans=None,
+    )
+
+
+@pytest.fixture
+def dummy_model() -> tuple:
+    """Fixture for a dummy model and processor."""
+    model_name = "facebook/wav2vec2-base-960h"
+    processor = Wav2Vec2Processor.from_pretrained(model_name)
+    model = Wav2Vec2ForCTC.from_pretrained(model_name)
+    return model, processor
+
+
+@pytest.fixture
+def script_line_fixture() -> ScriptLine:
+    """Pytest fixture to create a ScriptLine object.
+
+    Returns:
+        ScriptLine: An instance of ScriptLine.
+    """
+    data = {
+        "text": "test",
+        "speaker": "Speaker Name",
+        "start": 0.0,
+        "end": 10.0,
+        "chunks": [
+            {"text": "Chunk 1 text", "speaker": "Chunk 1 speaker", "start": 0.0, "end": 5.0},
+            {"text": "Chunk 2 text", "speaker": "Chunk 2 speaker", "start": 5.0, "end": 10.0},
+        ],
+    }
+    return ScriptLine.from_dict(data)
+
+
+def test_preprocess_segments() -> None:
+    """Test preprocessing of segments."""
+    transcript = [SingleSegment(start=0.0, end=1.0, text="test")]
+    model_dictionary = {"t": 0, "e": 1, "s": 2}
+    preprocessed_segments = _preprocess_segments(transcript, model_dictionary, "en", False, False)
+    assert preprocessed_segments[0]["clean_char"] == ["t", "e", "s", "t"]
+
+
+def test_can_align_segment(dummy_segment: SingleSegment) -> None:
+    """Test if a segment can be aligned."""
+    model_dictionary = {"t": 0, "e": 1, "s": 2}
+    assert _can_align_segment(dummy_segment, model_dictionary, 0.0, 10.0)
+
+
+def test_merge_repeats() -> None:
+    """Test merging of repeated tokens."""
+    path = [Point(0, 0, 1.0), Point(0, 1, 1.0), Point(1, 2, 1.0)]
+    transcript = "test"
+    segments = _merge_repeats(path, transcript)
+    assert len(segments) == 2
+
+
+def test_interpolate_nans() -> None:
+    """Test interpolation of NaN values."""
+    series = pd.Series([0.0, np.nan, 2.0])
+    interpolated_series = _interpolate_nans(series)
+    assert interpolated_series.isnull().sum() == 0
+
+
+if os.getenv("GITHUB_ACTIONS") != "true":
+
+    def test_get_prediction_matrix(dummy_model: tuple) -> None:
+        """Test generation of prediction matrix."""
+        model, _ = dummy_model
+        waveform_segment = torch.randn(1, 16000)
+        prediction_matrix = _get_prediction_matrix(model, waveform_segment, None, "huggingface", DeviceType.CPU)
```

Comment:
@fabiocat93 As we discussed before, since HFModel and TorchAudioModel don't have an instance variable for the model itself (Out of curiosity, why they don't they have this?), it makes more sense to pass the model as a `torch.nn.Module`. As far as I know `torch.nn.Module` doesn't have a direct way to tell whether a model is a torch audio model or a hugging face model, so for the sake of making the code more interpretable and reliable and because it required fewer changes, I decided to pass a string with the model type.

Which of the following approaches do you think is best?

Possible:
- Do nothing. Adds bulk to the function signatures, and makes the code a little rigid.

- Load the model in  `_get_prediction_matrix`. That would slow down performance since it would be loaded multiple times. 

- Add a factory. This might cause issues with Pydra that we discussed.

- Update the HFModel and TorchAudioModel types to store a reference to the loaded model and then pass those. This could be an optional instance variable to avoid causing problems with existing usage.

- Use intrinsic features of `torch.nn.Module` such as the solution below. This would be less transparent and reliable.
```
def _get_prediction_matrix(
    model: torch.nn.Module,
    waveform_segment: torch.Tensor,
    lengths: Optional[torch.Tensor],
    device: DeviceType,
) -> torch.Tensor:
    """Generate prediction matrix from the alignment model.

    Args:
        model (torch.nn.Module): The alignment model.
        waveform_segment (torch.Tensor): The audio segment to be processed.
        lengths (Optional[torch.Tensor]): Lengths of the audio segments.
        device (DeviceType): The device to run the model on.

    Returns:
        torch.Tensor: The prediction matrix.
    """
    with torch.inference_mode():
        if hasattr(model, 'forward') and 'torchaudio' in str(type(model)):
            emissions, _ = model(waveform_segment.to(device.value), lengths=lengths)
        elif hasattr(model, 'logits') or hasattr(model, 'forward') and 'transformers' in str(type(model)):
            emissions = model(waveform_segment.to(device.value)).logits
        else:
            raise NotImplementedError(f"Align model of type {type(model)} not supported.")

        emissions = torch.log_softmax(emissions, dim=-1)

    return emissions
```